### PR TITLE
Vladz

### PR DESCRIPTION
**Pull request by grzabrodskiy**
Merging latest changes, some improvements

### CODE REVIEW

File: First/Entities/FighterRecord.cs [hunk 1]
```diff
@@ -8,8 +8,21 @@ public class FighterRecord
...
-        public double Rank { get; set; }
-        public double PreviousRank { get; set; }
+        private double _rank;
+
+        public double Rank {
+            get { return _rank; }
+            set { PreviousRank = _rank; _rank = value; }
+            }
+
+        public double PreviousRank { get; private set; } // automatically preserved
```
Comment (azabrod1) [inline First/Entities/FighterRecord.cs:18]: You are setting previousRank in Rank but also have a setter in PreviousRank (line 18)? Why do you need both?
I am not sure how it works 100%, just asking maybe you know this better

Comment (grzabrodskiy) [reply First/Entities/FighterRecord.cs:18]: I have to use a private setter for Previous Rank to use it in the Rank setter. Alternatively, I could have a separate _previousRank private variable.

Comment (grzabrodskiy) [reply First/Entities/FighterRecord.cs:18]: set { **PreviousRank = _rank**; _rank = value; }
            }

        public double PreviousRank { get; private set; }

File: First/FightSimulation/EloFightSimulator.cs [hunk 2]
```diff
@@ -63,6 +62,10 @@ private static double eloDelta(double eloW, double eloL)
             return delta;
         }
 
+        private static double getNetworkViewers(double elo1, double elo2)
+        {
```
Comment (azabrod1) [inline First/FightSimulation/EloFightSimulator.cs:66]: Id consider exponential distributions as well

**Review by azabrod1 — APPROVED** [review]
ok

File: First/FightSimulation/EloFightSimulator.cs (hunk 2 above)
Comment (grzabrodskiy) [reply First/FightSimulation/EloFightSimulator.cs:66]: That's something Greg does

**Pull request merged**